frontend/src/credit_transfers/components/CreditTransferForm.js

```diff
@@ -78,69 +97,87 @@ const CreditTransferForm = props => (
         totalValue={props.totalValue}
       />
 
-      {Object.keys(props.errors).length > 0 &&
-        <Errors errors={props.errors} />
-      }
+      {Object.keys(props.errors).length > 0 && <Errors errors={props.errors} />}
 
-      {Object.keys(props.validationErrors).length > 0 &&
+      {Object.keys(props.validationErrors).length > 0 && (
         <Errors errors={props.validationErrors} />
-      }
+      )}
+      <p className="action-context-menu-available">Agreement Date (required)</p>
+    <div className="agreementDate">
+    <h3>
+        Date on which the written agreement to transfer credits was reached
+        between the suppliers:
+      </h3>
+      <div>
+        <label>Agreement Date:</label>
+        <input
+          className="form-control form-date"
+          type="date"
+          value={props.fields.agreementDate}
+          name='agreementDate'
+          onChange={props.handleInputChange}
+          required
+        />
+      </div>
+    </div>
 
-      <CreditTransferVisualRepresentation
-        creditsFrom={props.creditsFrom}
-        creditsTo={props.creditsTo}
-        loggedInUser={props.loggedInUser}
-        numberOfCredits={props.fields.numberOfCredits}
-        totalValue={props.totalValue}
-        tradeType={props.fields.tradeType}
-        zeroDollarReason={props.zeroDollarReason}
+      <CreditTransferCommentForm
+        isCommentingOnUnsavedCreditTransfer={props.id === 0}
+        isCreatingPrivilegedComment={false}
+        handleCommentChanged={props.handleCommentChanged}
+        embedded
       />
-
-      {(props.loggedInUser.hasPermission(PERMISSIONS_CREDIT_TRANSACTIONS.SIGN)) &&
+      {props.loggedInUser.hasPermission(
+        PERMISSIONS_CREDIT_TRANSACTIONS.SIGN
+      ) && (
         <CreditTransferTerms
           addToFields={props.addToFields}
           fields={props.fields}
           signingAuthorityAssertions={props.signingAuthorityAssertions}
           toggleCheck={props.toggleCheck}
         />
-      }
+      )}
 
-      <CreditTransferCommentForm
+      {/* <CreditTransferCommentForm
```

Comment:
If this was moved, let's just remove the commented code here.

Reply:
done